src/improvements/tasks/MultisigTask.sol

```diff
@@ -392,33 +415,51 @@ abstract contract MultisigTask is Test, Script {
         bytes32 hash = keccak256(getDataToSign(_childMultisig, approveCalldata));
         signatures = Signatures.prepareSignatures(_childMultisig, hash, signatures);
 
-        bool success = IGnosisSafe(_childMultisig).execTransaction(
-            MULTICALL3_ADDRESS,
-            0,
-            approveCalldata,
-            Enum.Operation.DelegateCall,
-            0,
-            0,
-            0,
-            address(0),
-            payable(address(0)),
-            signatures
-        );
+        if (vm.isContext(VmSafe.ForgeContext.ScriptBroadcast)) {
+            /// only broadcast if we are running forge script with the --broadcast flag
+            vm.broadcast();
+        }
 
-        require(success, "MultisigTask: approval failed");
+        execTransaction(_childMultisig, MULTICALL3_ADDRESS, 0, approveCalldata, Enum.Operation.DelegateCall, signatures);
     }
 
     /// @notice Executes the task with the given signatures.
     /// @param signatures The signatures to execute the task.
-    function execute(bytes memory signatures) public {
+    function execute(bytes memory signatures) public returns (VmSafe.AccountAccess[] memory) {
         bytes memory data = getCalldata();
         bytes32 hash = getHash();
 
-        signatures = Signatures.prepareSignatures(parentMultisig, hash, signatures);
+        /// if no signatures are attached, create them
+        if (signatures.length == 0) {
+            signatures = prepareSignatures(parentMultisig, hash);
+        } else {
+            /// otherwise we assume the signatures are already ordered
+            Signatures.sortUniqueSignatures(
+                parentMultisig, signatures, hash, IGnosisSafe(parentMultisig).getThreshold(), signatures.length
+            );
+        }
```

Comment:
Can you clarify the if/else block here in the code comments? I don't understand why "creating signatures" means? Also the `else` block says "we assume the signatures are already ordered" but then proceeds to sort the signatures anyway